### PR TITLE
Fix lstsq FutureWarning in NumpyMatrixOperator.apply_inverse

### LINKED ISSUE #1242: Future warning for least squares in NumpyMatrixOperator
Using `NumpyMatrixOperator` with Python 3.8 gives a future warning on the `rcond` parameter of `np.linalg.lstsq`

```
pymor/src/pymor/operators/numpy.py:322: FutureWarning: `rcond` parameter will change to the default of machine precision times ``max(M, N)`` where M and N are the input matrix dimensions.
To use the future default and silence this warning we advise to pass `rcond=None`, to keep using the old, explicitly pass `rcond=-1`.
  R, _, _, _ = np.linalg.lstsq(self.matrix, V.to_numpy().T)
```

### PR DESCRIPTION
**Pull request by sdrave**
Fixes #1242.

### CODE REVIEW

**Comment by TiKeil** [comment]
Looking, good but should probably also be changed in [operators/constructions.py](https://github.com/pymor/pymor/blob/main/src/pymor/operators/constructions.py#L817), in [algorithms/genericsolvers.py](https://github.com/pymor/pymor/blob/main/src/pymor/algorithms/genericsolvers.py#L352) and in the [tests](https://github.com/pymor/pymor/blob/main/src/pymortests/operators.py#L425)?

File: src/pymor/operators/numpy.py [hunk 1]
```diff
@@ -319,7 +319,12 @@ def apply_inverse(self, V, mu=None, initial_guess=None, least_squares=False,
         else:
             if least_squares:
                 try:
-                    R, _, _, _ = np.linalg.lstsq(self.matrix, V.to_numpy().T)
+                    import pymor.config
+                    if tuple(int(v) for v in pymor.config.NUMPY_VERSION.split('.')) >= (1, 14, 0):
```
Comment (renefritze) [inline src/pymor/operators/numpy.py:323]:
```suggestion
                    from packaging.version import parse
                    if parse(pymor.config.NUMPY_VERSION) >= parse('1.14'):
```

**Review by renefritze — CHANGES_REQUESTED** [review]

File: src/pymor/operators/numpy.py (hunk 1 above)
Comment (sdrave) [reply src/pymor/operators/numpy.py:323]: Ah, I knew we had a package for comparing versions, but I was to lazy too look up its name. Thanks ..

**Comment by pmli** [comment]
Looking at `requrements.txt`, the oldest supported NumPy version is 1.15.4. @renefritze, could you confirm? In that case, it's not necessary to check the NumPy version.

> Looking good but should probably also be changed in [operators/constructions.py](https://github.com/pymor/pymor/blob/main/src/pymor/operators/constructions.py#L817), in [algorithms/genericsolvers.py](https://github.com/pymor/pymor/blob/main/src/pymor/algorithms/genericsolvers.py#L352) and in the [tests](https://github.com/pymor/pymor/blob/main/src/pymortests/operators.py#L425)?

I agree that it should probably also be changed in `VectorOperator.apply_inverse`. In `lgmres`, `lstsq` from SciPy is used, so maybe one option is to use it elsewhere. For the tests, I'm not sure if warnings are interpreted as errors, but it might be good to remove them.

**Comment by renefritze** [comment]
> Looking at `requrements.txt`, the oldest supported NumPy version is 1.15.4. @renefritze, could you confirm? 

That is indeed correct, yes.

**Comment by sdrave** [comment]
I removed the branch for older numpy's and also applied `rcond=None` in all other places where `numpy.linalg.lstsq` has been used. Note that in `genericsolvers` the code (which originates from scipy) uses the scipy version of `lstsq` which has another signature.

**Review by renefritze — APPROVED** [review]

**Comment by sdrave** [comment]
@renefritze, there are two failing jobs due to hanging MPI processes as it seems. This seems new?

**Comment by renefritze** [comment]
> @renefritze, there are two failing jobs due to hanging MPI processes as it seems. This seems new?

It's not. It's symptom/occurrence of the same issue(s) I mentioned on gitter. Just restart for now please.

**Pull request merged**